data-collection/src/main/java/refactoringml/RefactoringAnalyzer.java

```diff
@@ -33,49 +29,52 @@
 		this.project = project;
 		this.db = db;
 		this.repository = repository;
 		this.storeFullSourceCode = storeFullSourceCode;
-
 		this.tempDir = null;
+		this.pmDatabase = pmDatabase;
 		this.fileStorageDir = lastSlashDir(fileStorageDir);
 	}
 
-	public List<RefactoringCommit> collectCommitData(RevCommit commit, CommitMetaData superCommitMetaData, List<Refactoring> refactoringsToProcess) throws IOException {
+	public List<RefactoringCommit> collectCommitData(RevCommit commit, CommitMetaData superCommitMetaData, List<Refactoring> refactoringsToProcess, List<DiffEntry> entries) {
 		List<RefactoringCommit> allRefactorings = new ArrayList<>();
 
 		try {
-			// get the map between new path -> old path
-			HashMap<String, String> filesMap = getMapWithOldAndNewFiles(repository, commit);
+			//TODO: is this obsolete? JGit misses many files, e.g. after move source dir refactorings
+			//get the map between new path -> old path
+			HashMap<String, String> filesMap = getMapWithOldAndNewFiles(entries);
 
 			// get the map between class names
 			HashMap<String, String> classAliases = getClassAliases(refactoringsToProcess);
-
 			//Iterate over all Refactorings found for this commit
 			for (Refactoring refactoring : refactoringsToProcess) {
+
 				String refactoringSummary = refactoring.toString().trim();
 				log.debug("Process Commit [" + commit.getId().getName() + "] with Refactoring: [" + refactoringSummary + "]");
 
 				//loop over all refactored classes, multiple classes can be refactored by the same refactoring, e.g. Extract Interface Refactoring
 				for (ImmutablePair<String, String> pair : refactoredFilesAndClasses(refactoring, refactoring.getInvolvedClassesBeforeRefactoring())) {
 					// get the name of the file before the refactoring
 					// if the one returned by RMiner exists in the map, we use the one in the map instead
-					String refactoredClassFile = pair.getLeft();
-					if(filesMap.containsKey(refactoredClassFile))
-						refactoredClassFile = filesMap.get(refactoredClassFile);
+					String refactoredClassFile = enforceUnixPaths(pair.getLeft());
+					//ignore the filename from jgit for move source dirs etc, because it is unreliable (#133)
+					if(!isClassRename(refactoring) && filesMap.containsKey(refactoredClassFile))
+						refactoredClassFile = enforceUnixPaths(filesMap.get(refactoredClassFile));
 
 					/**
 					 * Sometimes, RMiner finds refactorings in newly introduced classes. Often, as part of larger refactorings.
 					 * (See https://github.com/tsantalis/RefactoringMiner/issues/89 for a better understanding)
 					 * We can't use those, as we need a file in the previous commit to collect metrics.
 					 * Thus, we skip this refactoring.
 					 */
-					if(fileDoesNotExist(refactoredClassFile)) {
+					if(!isClassRename(refactoring) && fileDoesNotExist(refactoredClassFile)) {
```

Comment:
Why do we need this `!isClassRename(refactoring)` check here?

Maybe removing it would be the tool more resilient. If if faces another corner case we can't identify, it will log it here and move to the next one, without having some extra behaviour we don't understand.

Reply:
I added this as a saveguard, but you are right we should handle this differently.